specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json

```diff
@@ -316,6 +316,58 @@
         "connectionString"
       ]
     },
+    "AzureSqlMILinkedService": {
+      "x-ms-discriminator-value": "AzureSqlMI",
+      "description": "Azure SQL Managed Instance linked service.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/LinkedService"
+        }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "Azure SQL Managed Instance linked service properties.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AzureSqlMILinkedServiceTypeProperties"
+        }
+      },
+      "required": [
+        "typeProperties"
+      ]
+    },
+    "AzureSqlMILinkedServiceTypeProperties": {
+      "description": "Azure SQL Managed Instance linked service properties.",
+      "properties": {
+        "connectionString": {
+          "description": "The connection string. Type: string, SecureString or AzureKeyVaultSecretReference.",
+          "type": "object"
```

Comment:
Why isn't it marked as `string`? Same for similar properties.

Reply:
As mentioned in the description, this property can be any of String, SecureString or AzureKeyVaultSecretReference.  That's why its type is object.